addons/l10n_ec/models/account_move.py

```diff
@@ -0,0 +1,259 @@
+        'ec_dt_42',
+        'ec_dt_43',
+        'ec_dt_45',
+        'ec_dt_47',
+        'ec_dt_48'
+    ],
+    "02": [
+        'ec_dt_03',
+        'ec_dt_04',
+        'ec_dt_05',
+        'ec_dt_09',
+        'ec_dt_19',
+        'ec_dt_41',
+        'ec_dt_294',
+        'ec_dt_344'
+    ],
+    "03": [
+        'ec_dt_03',
+        'ec_dt_04',
+        'ec_dt_05',
+        'ec_dt_09',
+        'ec_dt_15',
+        'ec_dt_19',
+        'ec_dt_41',
+        'ec_dt_45',
+        'ec_dt_294',
+        'ec_dt_344'
+    ],
+    "04": [
+        'ec_dt_04',
+        'ec_dt_05',
+        'ec_dt_18',
+        'ec_dt_41',
+        'ec_dt_44',
+        'ec_dt_47',
+        'ec_dt_48',
+        'ec_dt_49',
+        'ec_dt_50',
+        'ec_dt_51',
+        'ec_dt_52',
+        'ec_dt_370',
+        'ec_dt_371',
+        'ec_dt_372',
+        'ec_dt_373'
+    ],
+    "05": [
+        'ec_dt_04',
+        'ec_dt_05',
+        'ec_dt_18',
+        'ec_dt_41',
+        'ec_dt_44',
+        'ec_dt_47',
+        'ec_dt_48',
+        'ec_dt_370',
+        'ec_dt_371',
+        'ec_dt_372',
+        'ec_dt_373'
+    ],
+    "06": [
+        'ec_dt_04',
+        'ec_dt_05',
+        'ec_dt_18',
+        'ec_dt_41',
+        'ec_dt_44',
+        'ec_dt_47',
+        'ec_dt_48',
+        'ec_dt_370',
+        'ec_dt_371',
+        'ec_dt_372',
+        'ec_dt_373'
+    ],
+    "07": [
+        'ec_dt_04',
+        'ec_dt_05',
+        'ec_dt_18'
+    ],
+    "09": [
+        'ec_dt_01',
+        'ec_dt_04',
+        'ec_dt_05',
+        'ec_dt_15',
+        'ec_dt_16',
+        'ec_dt_41',
+        'ec_dt_47',
+        'ec_dt_48',
+    ],
+    "20": [
+        'ec_dt_01',
+        'ec_dt_04',
+        'ec_dt_05',
+        'ec_dt_15',
+        'ec_dt_16',
+        'ec_dt_41',
+        'ec_dt_47',
+        'ec_dt_48'
+    ],
+    "21": [
+        'ec_dt_01',
+        'ec_dt_04',
+        'ec_dt_05',
+        'ec_dt_15',
+        'ec_dt_16',
+        'ec_dt_41',
+        'ec_dt_47',
+        'ec_dt_48'
+    ],
+}
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    l10n_ec_sri_payment_id = fields.Many2one(
+        comodel_name="l10n.ec.sri.payment",
+        string="Payment Method (SRI)",
+    )
+
+    l10n_ec_emission_type = fields.Selection(
+        string="Emission type",
+        selection=[
+            ("pre_printed", "Pre Printed"),
+            ("auto_printer", "Auto Printer"),
+            ("electronic", "Electronic"),
+        ],
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        default="electronic",
+    )
+
+    def _get_l10n_ec_identification_type(self):
+        self.ensure_one()
+        move = self
+        _valid, l10n_ec_type_sri = move.partner_id.commercial_partner_id.check_vat_ec()
+        l10n_ec_is_exportation = move.partner_id.commercial_partner_id.country_id.code != 'EC'
+        identification_code = False
+        if move.move_type in ("in_invoice", "in_refund"):
```

Comment:
Putting an l10n_ec_code on the identification type might not be a bad idea either.

Reply:
Ok, but in this case the value depends on the transaction where I am located at that moment, being in latam document type, it would not allow me to distinguish the context well, since a type can be used for purchases and sales, which makes me have this separate logic

Reply:
If the RUC is wrong > warning that the number is incorrect > go and check the number/identification type